src/App.jsx

```diff
@@ -14,45 +16,107 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_BY_ALPHABET = 'sortByAlphabet';
+const SORT_BY_LENGTH = 'sortByLength';
+
+const sortGoods = (sortBy) => {
+  const copy = [...goodsFromServer];
+
+  switch (sortBy) {
+    case SORT_BY_ALPHABET:
+      return copy.sort((good1, good2) => good1.localeCompare(good2));
+
+    case SORT_BY_LENGTH:
+      return copy.sort((good1, good2) => good1.length - good2.length);
+
+    case '':
+      return copy;
+
+    default:
+      return 0;
+  }
+};
+
+export const App = () => {
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortBy, setSortBy] = useState('');
+  const [goods, setGoods] = useState([...goodsFromServer]);
+
+  const resetSorting = () => {
+    setIsReversed(false);
+    setSortBy('');
+    setGoods([...goodsFromServer]);
+  };
+
+  const sortOnClick = (parameter) => {
+    setSortBy(parameter);
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+    const sortedGoods = sortGoods(parameter);
+
+    setGoods(sortedGoods);
+  };
+
+  const setAscOrder = () => {
+    sortOnClick(sortBy);
+    setIsReversed(!isReversed);
+  };
+
+  if (isReversed) {
+    goods.reverse();
+  }
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button', ['is-info'], {
```

Comment:
Please fix for all cases
```suggestion
          className={cn('button is-info', {
```